docs/examples/plot_custom_parser.py

```diff
@@ -0,0 +1,286 @@
+++++++++++++++++
+
+To do that, we implement a new estimator which is identical
+which trains whichever classifier we need.
+"""
+import inspect
+import numpy as np
+import skl2onnx
+import onnxruntime
+import onnx
+import sklearn
+from sklearn.base import ClassifierMixin, BaseEstimator, clone
+from sklearn.datasets import load_iris
+from sklearn.linear_model import LogisticRegression
+from sklearn.model_selection import train_test_split
+from skl2onnx import update_registered_converter
+import os
+from onnx.tools.net_drawer import GetPydotGraph, GetOpNodeProducer
+import onnxruntime as rt
+from onnxconverter_common.onnx_ops import (
+    apply_identity, apply_cast, apply_greater
+)
+from skl2onnx import to_onnx, get_model_alias
+from skl2onnx.proto import onnx_proto
+from skl2onnx.common._registration import get_shape_calculator
+from skl2onnx.common.data_types import FloatTensorType, Int64TensorType
+import matplotlib.pyplot as plt
+
+
+class ValidatorClassifier(BaseEstimator, ClassifierMixin):
+
+    def __init__(self, estimator=None, threshold=0.75):
+        ClassifierMixin.__init__(self)
+        BaseEstimator.__init__(self)
+        if estimator is None:
+            estimator = LogisticRegression(solver='liblinear')
+        self.estimator = estimator
+        self.threshold = threshold
+
+    def fit(self, X, y, sample_weight=None):
+        sig = inspect.signature(self.estimator.fit)
+        if 'sample_weight' in sig.parameters:
+            self.estimator_ = clone(self.estimator).fit(
+                X, y, sample_weight=sample_weight)
+        else:
+            self.estimator_ = clone(self.estimator).fit(X, y)
+        return self
+
+    def predict(self, X):
+        return self.estimator_.predict(X)
+
+    def predict_proba(self, X):
+        return self.estimator_.predict_proba(X)
+
+    def validate(self, X):
+        pred = self.predict_proba(X)
+        mx = pred.max(axis=1)
+        return (mx >= self.threshold) * 1
+
+
+data = load_iris()
+X, y = data.data, data.target
+X_train, X_test, y_train, y_test = train_test_split(X, y)
+
+model = ValidatorClassifier()
+model.fit(X_train, y_train)
+
+##############################
+# Let's now measure the indicator which tells
+# the probability of the prediction is above
+# a threshold.
+
+print(model.validate(X_test))
+
+####################################
+# Conversion to ONNX
+# +++++++++++++++++++
+#
+# The conversion fails for a new model because
+# the library does not any converter associated
+# to this new model.
+
+try:
+    to_onnx(model, X_train[:1].astype(np.float32))
+except RuntimeError as e:
+    print(e)
+
+############################################
+# Custom converter
+# ++++++++++++++++
+#
+# We reuse some pieces of code from :ref:`l-custom-model`.
+# The shape calculator defines the shape of every output
+# of the converted model.
+
+
+def validator_classifier_shape_calculator(operator):
+
+    input = operator.inputs[0]  # inputs in ONNX graph
```

Comment:
input is a build-in function in Python, I would suggest you to use a different name.

Reply:
fixed